File: client/GhostClient/Core/Network/Packets/SpawnEntityPacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using GhostClient.Core;

namespace Ghost.Core.Network.Packets
{
    public class SpawnEntityPacket : Packet
    {
        public SpawnEntityPacket(Stream stream) : base(stream)
        {
        }

        public SpawnEntityPacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public SpawnEntityPacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            short type = Consume(2).ShortValue;
            short id = Consume(2).ShortValue;

            byte nameLength = Consume(1).ByteValue;
            string name = Consume(nameLength).AsciiStringValue;

            float x = Consume(4).FloatValue;
            float y = Consume(4).FloatValue;

            double angle = Consume(8).DoubleValue;

            GameHandler.Game.SpawnEntity(type, id, name, x, y, angle);
        }
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/SpawnEntityPacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient

class SpawnEntityPacket : Packet<PlayerClient>() {

    override fun handle() {
        val type : Short = consume(2).asShort()
        val id : Short = consume(2).asShort()

        val nameLength = consume(4).asInt()
        val name : String = consume(nameLength).asString()

        val x : Float = consume(4).asFloat()
        val y : Float = consume(4).asFloat()

        val angle : Double = consume(8).asDouble()

        val width = consume(2).asShort()
        val height = consume(2).asShort()

        client.game.spawn(type, id, name, x, y, angle, width, height)
    }
}

